### PR TITLE
Field Folder, Update README.md, Battle overlay progress

### PR DESCRIPTION
**Pull request by adrienntindall**

### CODE REVIEW

File: include/constants/battle.h [hunk 1]
```diff
@@ -163,7 +176,9 @@ enum Terrain {
 #define STATUS2_MEAN_LOOK                   (1 << 26)
```
Comment (red031000) [inline include/constants/battle.h:176]: I'm going to be irritating and ask if there's another name that can be used other than STATUS2

File: src/battle/overlay_12_0224E4FC.c [hunk 2]
```diff
@@ -1962,3 +1965,920 @@ void ov12_02251710(BattleSystem *bsys, BATTLECONTEXT *ctx) {
...
+                }
+            }
+            i++;
+        } while (sTypeEffectiveness[i][0] != TYPE_ENDTABLE);
+    }
+    
+    if (CheckBattlerAbilityIfNotIgnored(ctx, battlerIdAttacker, battlerIdTarget, ABILITY_WONDER_GUARD) == TRUE && ov12_02258440(ctx, moveNo) &&
+        (!(*moveStatusFlags & MOVE_STATUS_SUPER_EFFECTIVE) || ((*moveStatusFlags & MOVE_STATUS_ANY_EFFECTIVE) == MOVE_STATUS_ANY_EFFECTIVE)) && movePower) {
+        *moveStatusFlags |= MOVE_STATUS_WONDER_GUARD_IMMUNE;
+    } else if (!(ctx->linkStatus & (1 << 11)) && !(ctx->linkStatus & (1 << 15))) {
```
Comment (red031000) [inline src/battle/overlay_12_0224E4FC.c:2182]: constants? though I think you mentioned that you were going to do the link status one later... not 100% sure though

File: src/battle/overlay_12_0224E4FC.c [hunk 3]
```diff
@@ -1962,3 +1965,920 @@ void ov12_02251710(BattleSystem *bsys, BATTLECONTEXT *ctx) {
...
+        }
+    }
+    
+    return ret;
+}
+
+#ifdef NONMATCHING
+//FIXME: Matches up to regswaps: https://decomp.me/scratch/h06Ji -adrienn
+int ov12_02251D28(BattleSystem *bsys, BATTLECONTEXT *ctx, int moveNo, int moveTypeDefault, int battlerIdAttacker, int battlerIdTarget, int damage, u32 *moveStatusFlags) {
+    int i;
```
Comment (red031000) [inline src/battle/overlay_12_0224E4FC.c:2115]: I personally don't really like the idea of declaring all the vars at the top of a function like this.... looks kinda ugly.... if possible, would you be able to declare them inline with assignments (or as late as possible at least) (provided it still matches, that is)

File: src/battle/overlay_12_0224E4FC.c [hunk 4]
```diff
@@ -1962,3 +1965,920 @@ void ov12_02251710(BattleSystem *bsys, BATTLECONTEXT *ctx) {
...
+        moveType = moveTypeDefault;
+    } else {
+        moveType = ctx->unk_334.moveData[moveNo].type;
+    }
+    
+    if (abilityAttacker != ABILITY_MOLD_BREAKER && abilityTarget == ABILITY_LEVITATE && moveType == TYPE_GROUND && !(ctx->fieldCondition & FIELD_CONDITION_GRAVITY) && item != HOLD_EFFECT_SPEED_DOWN_GROUNDED) {
+        *moveStatusFlag |= MOVE_STATUS_NO_EFFECT;
+    } else {
+        i = 0;
+        do {
```
Comment (red031000) [inline src/battle/overlay_12_0224E4FC.c:2620]: this feels like it should be a for loop

File: src/battle/overlay_12_0224E4FC.c [hunk 5]
```diff
@@ -1962,3 +1965,920 @@ void ov12_02251710(BattleSystem *bsys, BATTLECONTEXT *ctx) {
...
+        break;
+    }
+    
+    return cnt;
+}
+
+//FIXME: Function name is wrong
+BOOL BattleCtx_IsIdenticalToCurrentMove(BATTLECONTEXT *ctx, int moveNo) {
+    switch (ctx->unk_334.moveData[moveNo].effect) {
+    case 26:
```
Comment (red031000) [inline src/battle/overlay_12_0224E4FC.c:2824]: these also should probably be constants

**Review by red031000 — CHANGES_REQUESTED** [review]

File: include/constants/battle.h [hunk 6]
```diff
@@ -90,7 +90,18 @@ enum Terrain {
...
+//Move status
+#define MOVE_STATUS_MISS                    (1 << 0)
+#define MOVE_STATUS_SUPER_EFFECTIVE         (1 << 1) 
+#define MOVE_STATUS_NOT_VERY_EFFECTIVE      (1 << 2)
+#define MOVE_STATUS_NO_EFFECT               (1 << 3) 
+#define MOVE_STATUS_LEVITATE_IMMUNE         (1 << 11)
+#define MOVE_STATUS_WONDER_GUARD_IMMUNE     (1 << 18) 
+#define MOVE_STATUS_MAGNET_RISE_IMMUNE      (1 << 20)
+
+#define MOVE_STATUS_ANY_EFFECTIVE           (MOVE_STATUS_SUPER_EFFECTIVE | MOVE_STATUS_NOT_VERY_EFFECTIVE)
```
[the remark this answers is not in the document]
Comment (adrienntindall) [reply include/constants/battle.h:104]: eh I got nothing

File: src/battle/overlay_12_0224E4FC.c [hunk 7]
```diff
@@ -1962,3 +1965,920 @@ void ov12_02251710(BattleSystem *bsys, BATTLECONTEXT *ctx) {
...
+    return nonSelectableMoves;
+}
+
+//Buffer messages related to being unable to select moves?
+BOOL ov12_02251A28(BattleSystem *bsys, BATTLECONTEXT *ctx, int battlerId, int movePos, BATTLEMSG *msg) {
+    BOOL ret = TRUE;
+    
+    if (StruggleCheck(bsys, ctx, battlerId, 0, STRUGGLE_CHECK_DISABLED) & MaskOfFlagNo(movePos)) {
+        msg->tag = 10;
+        msg->id = 609;
```
[the remark this answers is not in the document]
Comment (adrienntindall) [reply src/battle/overlay_12_0224E4FC.c:2019]: I couldn't find it when I looked but I'll take Blu's suggestion and see if they line up with the ones in 197

File: src/battle/overlay_12_0224E4FC.c (hunk 3 above)
Comment (adrienntindall) [reply src/battle/overlay_12_0224E4FC.c:2115]: thing is the variable declaration order effects matching a lot of the time due to how the compiler decides to put things in registries, so having them at the top like this is the easiest way to ensure matching

File: src/battle/battle_command.c [hunk 8]
```diff
@@ -4075,7 +4075,7 @@ BOOL BtlCmd_TryImprison(BattleSystem *bsys, BATTLECONTEXT *ctx) {
 }
 #else
 asm BOOL BtlCmd_TryImprison(BattleSystem *bsys, BATTLECONTEXT *ctx) {
-    	push {r3, r4, r5, r6, r7, lr}
+    push {r3, r4, r5, r6, r7, lr}
 	sub sp, #0x18
```
[the remark this answers is not in the document]
Comment (adrienntindall) [reply src/battle/battle_command.c:4079]: my bad, they're tabs so they appeared to be on the same level locally

File: include/constants/battle.h (hunk 1 above)
Comment (adrienntindall) [reply include/constants/battle.h:176]: volatile is for burn etc, not sure what the other status effect terminology is unless you want STATUS_NON_VOLITILE_MEAN_LOOK everywhere which is arguably more annoying

File: include/constants/battle.h [hunk 9]
```diff
@@ -172,6 +187,18 @@ enum Terrain {
...
+//Struggle Checks
+#define STRUGGLE_CHECK_NO_MOVES             (1 << 0)
+#define STRUGGLE_CHECK_NO_PP                (1 << 1)
+#define STRUGGLE_CHECK_DISABLED             (1 << 2)
+#define STRUGGLE_CHECK_TORMENT              (1 << 3)
+#define STRUGGLE_CHECK_TAUNT                (1 << 4)
+#define STRUGGLE_CHECK_IMPRISON             (1 << 5)
+#define STRUGGLE_CHECK_GRAVITY              (1 << 6)
+#define STRUGGLE_CHECK_HEAL_BLOCK           (1 << 7)
+#define STRUGGLE_CHECK_ENCORE               (1 << 8) //unused because they straight up forgot
```
[the remark this answers is not in the document]
Comment (adrienntindall) [reply include/constants/battle.h:199]: literally 💀

File: include/constants/battle.h (hunk 1 above)
Comment (red031000) [reply include/constants/battle.h:176]: ehhh note to self: come back to this later

**Review by red031000 — APPROVED** [review]

**Pull request merged**